src/spirc.rs

```diff
@@ -170,6 +172,16 @@ impl<D: SpircDelegate> SpircManager<D> {
             protocol::spirc::MessageType::kMessageTypePause => {
                 self.delegate.pause();
             }
+            protocol::spirc::MessageType::kMessageTypeNext => {
+            	self.index = (self.index + 1) % self.tracks.len() as u32;
+                let track = self.tracks[self.index as usize];
```

Comment:
What do you mean by no way to determine the queue order?

I assume the tracks in above tracklist are in the order a client would play them. 9 were already played, the 10th is playing (`playing_track_index: 9`).

The two manually queued tracks are inserted after the currently playing track, with the `queued` flag (=by user). All other tracks were automatically added (=by app) when I first tapped on a song. (It passively adds all tracks past the one you clicked of the same album or playlist. They have a lower priority than manually queued tracks.)

![spotify-android-queue](https://cloud.githubusercontent.com/assets/1309249/12354416/169ac5c8-bb96-11e5-8d23-2c43b260c410.png)

Debug printing is really simple, all I did is to match replace messages and print the protobuf frame to console:

```rust
            protocol::spirc::MessageType::kMessageTypeReplace => {
                println!("{:?}", frame);
            }
```

I copied the output from console and edited is manually in Notepad++ to make it more readable. You could probably do that with some Rust code too, but I have zero knowledge about Rust.